### ISSUE TITLE
Custom pluralization

### ISSUE THREAD

**Issue by Eneroth3** [Eneroth3/ordbok#1]
Currently the library only supports the basic pluralization used in English, i.e. differentiating between the count "one" and "other". For instance "You have one message" and "You have 5 message**s**" .

In some languages this sentence would be needed to be written in more than two ways depending on the message count. In some languages the rule for what grammatically corresponds to one is different, e.g. that 21, 31, 41 etc use the same phrasing. as 1 (but 11 rarely does).

For this library to fully support more language, it would be useful if these rules could be specified in the language file similar to:
````
# Make every count ending with 1, but not ending with 11, grammatically correspond to 1.
pluralization_rule: {
  one: "x % 10 == 1 && x % 100 != 11"
}
````

**Comment by Eneroth3** [comment]
Specification for keys: http://www.unicode.org/cldr/charts/29/supplemental/language_plural_rules.html#comparison

**Comment by Eneroth3** [comment]
````ruby
# Using Russian rules for this example/experiment.
# Use pluralization from first rule that matched x.
#
# In CLDR :many and :other is accidentally swapped in Russian, but kept as it is
# too late to change.
# http://cldr.unicode.org/index/cldr-spec/plural-rules#TOC-Choosing-Plural-Category-Names
#
# This explains the somewhat awkward :many rule "true", which matches all x.
# Ideally the first rule, matching all floats, should have been labeled :many,
# and the numerics, now regarded as :many, should fail all tests and fall back to
# :other.
ru = {
  other: "x.is_a?(Float)",
  one: "x % 10 == 1 && x % 100 != 11",
  few: "(x % 10).between?(2, 4) && !(x % 100).between?(12, 14)",
  many: "true"
}

def pluralize(count, rules)
  rules.each_pair do |plural, rule|
    x = count
    return plural if eval(rule)
  end

  # Always fall back to :other if no rule matched count.
  :other
end

pluralize(1, ru)
# => 1

pluralize(3, ru)
# => :few

pluralize(3.4, ru)
# => :other
````

**Comment by Eneroth3** [comment]
5 different rules are implemented in 2c471f651dcc14fa83d7c176f784f4a890045dc3. The language files reference these by name as opposed to containing serialized executable code.

All languages supported by SketchUp should now be covered.
